### DISCUSSION TITLE
How to get start?

### DISCUSSION THREAD

**Discussion by InamulRehman** [microsoft/IoT-For-Beginners#280] [General]
Hi!
 I have learned c++ programming and I am improving my logic using c++. Also, I am learning data structures and algorithms in
 c ++ .
So can I contribute to this Open source project or can I learn something from here?
Any guide as I am a starter.
Thanks!

**Comment by Iftu119** [comment]
Hi @InamulRehman ! 

- **Contribution :** You can contribute to this curriculum as a translator. If you want to add translated files, please read the [translation guidelines](TRANSLATIONS.md) and add input [to one of the translations issues](https://github.com/microsoft/IoT-For-Beginners/issues?q=is%3Aissue+is%3Aopen+label%3Atranslation). If you want to translate into a new language, please raise a new issue for tracking. Also you can raise issues if there's any problem, discuss with others, especially the repository maintainers and then submit PR (as per the discussion)
- **Learning :** To use this curriculum on your own, fork the entire repo and complete the exercises on your own, starting with a pre-lecture quiz, then reading the lecture and completing the rest of the activities. Try to create the projects by comprehending the lessons rather than copying the solution code; however, that code is available in the /solutions folders in each project-oriented lesson. Another idea would be to form a study group with friends and go through the content together. For further study, it is recommended to use [Microsoft Learn](https://docs.microsoft.com/en-gb/users/jimbobbennett/collections/ke2ehd351jopwr?WT.mc_id=academic-17441-jabenn).

Wishing you luck! 🍀

**Comment by InamulRehman** [comment]
Alright, I am not getting how do I contribute to translation issues.

**Comment by InamulRehman** [comment]
is translation will be done by writing code?

**Reply by jlooper** [reply]
Translations are done by translating a lesson or other asset to your own language, as per the guide linked on the home page of this repo

**Comment by InamulRehman** [comment]
*Learning :*  I was contributing to the lesson Getting Started With IoT but
i was failed to proceed because of not successfully set up a virtual
environment.
So how do i proceed?

**Comment by GoodnessEzeh** [comment]
If I may ask: What IDE did you use?

**Comment by Amruta200** [comment]
Visual studio
